internal/pkg/aws/ecs/service.go

```diff
@@ -163,26 +151,29 @@ func (s *ServiceArn) parse() (*parsedServiceArn, error) {
 	}, nil
 }
 
+func (s *ServiceArn) String() string {
+	return fmt.Sprintf("arn:%s:%s:%s:%s:service/%s/%s", s.partition, EndpointsID, s.region, s.accountID, s.clusterName, s.name)
+}
+
 // ClusterArn returns the cluster arn.
 // For example: arn:aws:ecs:us-west-2:1234567890:service/my-project-test-Cluster-9F7Y0RLP60R7/my-project-test-myService-JSOH5GYBFAIB
 // will return arn:aws:ecs:us-west-2:1234567890:cluster/my-project-test-Cluster-9F7Y0RLP60R7
-func (s *ServiceArn) ClusterArn() (string, error) {
-	parsedArn, err := s.parse()
-	if err != nil {
-		return "", err
-	}
-	return fmt.Sprintf("arn:%s:%s:%s:%s:cluster/%s", parsedArn.partition, EndpointsID, parsedArn.region, parsedArn.accountID, parsedArn.clusterName), nil
+func (s *ServiceArn) ClusterArn() string {
+	return fmt.Sprintf("arn:%s:%s:%s:%s:cluster/%s", s.partition, EndpointsID, s.region, s.accountID, s.clusterName)
 }
 
 // ServiceName returns the service name.
 // For example: arn:aws:ecs:us-west-2:1234567890:service/my-project-test-Cluster-9F7Y0RLP60R7/my-project-test-myService-JSOH5GYBFAIB
 // will return my-project-test-myService-JSOH5GYBFAIB
-func (s *ServiceArn) ServiceName() (string, error) {
-	parsedArn, err := s.parse()
-	if err != nil {
-		return "", err
-	}
-	return parsedArn.name, nil
+func (s *ServiceArn) ServiceName() string {
+	return s.name
```

Comment:
any reason to not just export this on the struct as `s.ServiceName`? (same for cluster name)

Reply:
Becuase everything in that struct should be private and init by the Parse function.